### DISCUSSION TITLE
helm-controller service account not refreshing token

### DISCUSSION THREAD

**Discussion by nicogl** [fluxcd/flux2#2732] [Q&A] [answered]
I have fluxv2 version 0.27.3 running on an EKS cluster (in AWS) and today I got a notification message referring to the helm-controller.
This is an excerpt of the notification:

_We have identified applications running in one or more of your Amazon EKS clusters that are not refreshing service account tokens. Applications making requests to Kubernetes API server with expired tokens will fail. You can resolve the issue by updating your application and its dependencies to use newer versions of Kubernetes client SDK that automatically refreshes the tokens._

_What is the problem?_

_Kubernetes version 1.21 graduated BoundServiceAccountTokenVolume feature [1] to beta and enabled it by default. This feature improves security of service account tokens by requiring a one hour expiry time, over the previous default of no expiration. This means that applications that do not refetch service account tokens periodically will receive an HTTP 401 unauthorized error response on requests to Kubernetes API server with expired tokens. You can learn more about the BoundServiceAccountToken feature in EKS Kubernetes 1.21 release notes [2]._

Considering I'm using an old version of flux, I went through the fluxv2 releases changelogs after version 0.27.3 and didn't find anything that might fix the mentioned issue.

Is this a known issue? If so, was it fixed on the latest version? Otherwise, is it in the pipeline to be fixed?

Thanks!

**Comment by stefanprodan** [comment, marked as answer]
This has been fixed in https://github.com/fluxcd/helm-controller/pull/480 and will be included in the next Flux release.
